cmd/do-agent/run.go

```diff
@@ -51,7 +53,11 @@ func run(w metricWriter, l limiter, dec decorate.Decorator, g gatherer) {
 		dec.Decorate(mfs)
 		log.Debug("stats decorated in %s", time.Since(start))
 
-		err = w.Write(mfs)
+		aggregated, err := aggregate.Aggregate(mfs, aggregateSpec)
+		if err != nil {
+			log.Error("failed to aggregate metrics: %v", err)
+		}
+		err = w.Write(aggregated)
```

Comment:
should we still write if we fail to aggregate?

Reply:
I guess we should probably try to

Reply:
fixed, now we just send diagnostic info no this case.